cg_lims/EPPs/qc/sequencing_quality_checker.py

```
@@ -0,0 +1,108 @@
+import logging
+import sys
+from typing import List
+
+from cg_lims.EPPs.qc.sequencing_artifact_manager import SequencingArtifactManager
+from cg_lims.exceptions import LimsError
+from cg_lims.models.sequencing_metrics import SampleLaneSequencingMetrics
+from cg_lims.status_db_api import StatusDBAPI
+
+LOG = logging.getLogger(__name__)
+
+
+class SequencingQualityChecker:
+    READS_MIN_THRESHOLD = 1000
+
+    def __init__(
+        self, status_db_api: StatusDBAPI, artifact_manager: SequencingArtifactManager
+    ) -> None:
+        self.artifact_manager: SequencingArtifactManager = artifact_manager
+        self.status_db_api: StatusDBAPI = status_db_api
+
+        self.q30_threshold = self.artifact_manager.q30_threshold
+        self.flow_cell_name = self.artifact_manager.flow_cell_name
+
+        self.sample_lanes_in_metrics = set()
+        self.failed_sample_count = 0
+
+
+
+    def _track_sample_lane(self, metrics: SampleLaneSequencingMetrics) -> None:
+        sample_lane = (metrics.sample_internal_id, metrics.flow_cell_lane_number)
+        self.sample_lanes_in_metrics.add(sample_lane)
+
+    def _get_sequencing_metrics(self) -> List[SampleLaneSequencingMetrics]:
+        try:
+            return self.status_db_api.get_sequencing_metrics_for_flow_cell(
+                self.flow_cell_name
+            )
+        except LimsError as e:
```

This won't catch all exceptions from the cg client since it uses `raise_for_status` and does not catch the `HttpErrors`. I think the exceptions should be caught and the error logged already in the client, this is redundant. Need to patch the client in a separate PR

---

https://github.com/Clinical-Genomics/cg_lims/pull/416/commits/49f5441cddbf46d03e9a96a41eaa3d88fa2bc52e

---

https://github.com/Clinical-Genomics/cg_lims/pull/418